App/navigators/RootStack.tsx

```diff
@@ -37,6 +59,72 @@ const RootStack: React.FC = () => {
     navigation.navigate(Screens.Terms)
   }
 
+  const initAgent = async () => {
+    if (initAgentInProcess) {
+      return
+    }
+
+    Toast.show({
+      type: ToastType.Info,
+      text1: t('StatusMessages.InitAgent'),
+      position: 'bottom',
+    })
+
+    //Flag to protect the init process from being duplicated
+    setInitAgentInProcess(true)
+
+    //Remove before production
+    try {
+      const newAgent = new Agent(
+        {
+          label: 'Aries Bifold',
+          mediatorConnectionsInvite: Config.MEDIATOR_URL,
+          mediatorPickupStrategy: MediatorPickupStrategy.Implicit,
+          walletConfig: { id: 'wallet4', key: '123' },
+          autoAcceptConnections: true,
+          autoAcceptCredentials: AutoAcceptCredential.ContentApproved,
+          logger: new ConsoleLogger(LogLevel.trace),
+          indyLedgers,
+          connectToIndyLedgersOnStartup: false,
+        },
+        agentDependencies
+      )
+
+      const wsTransport = new WsOutboundTransport()
+      const httpTransport = new HttpOutboundTransport()
+
+      newAgent.registerOutboundTransport(wsTransport)
+      newAgent.registerOutboundTransport(httpTransport)
+
+      await newAgent.initialize()
+      setAgent(newAgent) // -> This will set the agent in the global provider
+      setAgentInitDone(true)
+      Toast.show({
+        type: ToastType.Success,
+        text1: 'Wallet initialized',
+        autoHide: true,
+        visibilityTime: 2000,
+        position: 'bottom',
+      })
+    } catch (e: unknown) {
+      Toast.show({
+        type: ToastType.Error,
+        text1: t('Global.Failure'),
+        text2: (e as Error)?.message || t('Error.Unknown'),
+        visibilityTime: 2000,
+        position: 'bottom',
+      })
+    }
+
+    setInitAgentInProcess(false)
```

Comment:
If we error out during the agent initialization we'll display the toast and then set initAgentInProcess to false, which in theory would allow it to create the agent again, which could lead to a perpetual loop, right?

Reply:
setting init agent to false doesn't trigger initializing the agent, the only thing that triggers init agent is "authenticated" flag